theano/gpuarray/basic_ops.py

```diff
@@ -297,6 +297,9 @@ def get_params(self, node):
         # Default implementation, suitable for most sub-classes.
         # To be necessarly overridden if a subclass uses a ParamsType
         # (see GpuCumOp, GpuAdvancedIncSubtensor1_dev20 for examples).
+        assert (self.params_type is gpu_context_type and
+                node.inputs and
+                isinstance(node.inputs[0].type, GpuArrayType))
```

Comment:
why this condition? Maybe one GPU op would have many inputs and not all of them on the GPU. This would force the first inputs to be on the GPU. Why add this constrain?